packages/svelte/src/internal/client/reactivity/props.js

```diff
@@ -175,10 +175,13 @@ export function prop(props, key, flags, initial) {
 	// intermediate mode — prop is written to, but the parent component had
 	// `bind:foo` which means we can just call `$$props.foo = value` directly
 	if (setter) {
-		return function (/** @type {V} */ value) {
+		return function (/** @type {V} */ value, mutation = false) {
 			if (arguments.length === 1) {
 				/** @type {Function} */ (setter)(value);
 				return value;
+			} else if (mutation) {
+				/** @type {Function} */ (setter)(getter());
+				return value;
```

Comment:
what if instead of adding code to `prop`, we limited the blast radius to legacy components like this?

```diff
-$.bind_value(input, () => value().name, ($$value) => value(value().name = $$value, true));
+$.bind_value(input, () => value().name, ($$value) => value((value().name = $$value, value())));
```

Reply:
The `mutation: true` parameter is used for the "can get temporary out of sync" case further below, too, and we would only save the `setter(getter)` code, which minifies to almost nothing - I don't think that's worth the more complicated setup.